api/all/src/main/java/io/opentelemetry/api/trace/TraceId.java

```diff
@@ -44,84 +58,107 @@ public static String getInvalid() {
-   * as big-endian.
+   * <p>This is equivalent to calling {@link #fromBytes(byte[])} with the specified values stored as
+   * big-endian.
    *
-   * @param idHi the higher part of the {@code TraceId}.
-   * @param idLo the lower part of the {@code TraceId}.
+   * @param traceIdLongHigh the higher part of the long values representation of the {@code
+   *     TraceId}.
+   * @param traceIdLongLow the lower part of the long values representation of the {@code TraceId}.
+   * @return the lowercase hex (base16) representation of the {@code TraceId}.
    */
-  public static String fromLongs(long idHi, long idLo) {
+  public static String fromLongs(long traceIdLongHigh, long traceIdLongLow) {
+    if (traceIdLongHigh == 0 && traceIdLongLow == 0) {
+      return getInvalid();
+    }
     char[] chars = getTemporaryBuffer();
-    BigendianEncoding.longToBase16String(idHi, chars, 0);
-    BigendianEncoding.longToBase16String(idLo, chars, 16);
+    BigendianEncoding.longToBase16String(traceIdLongHigh, chars, 0);
+    BigendianEncoding.longToBase16String(traceIdLongLow, chars, 16);
     return new String(chars);
   }
 
   /**
-   * Returns the rightmost 8 bytes of the trace-id as a long value. This is used in
-   * ProbabilitySampler.
+   * Returns the rightmost 8 bytes of the trace-id as a long value. This is used in {@code
+   * TraceIdRatioBasedSampler}.
    *
    * <p>This method is marked as internal and subject to change.
    *
    * @return the rightmost 8 bytes of the trace-id as a long value.
    */
   public static long getTraceIdRandomPart(CharSequence traceId) {
-    return traceIdLowBytesAsLong(traceId);
+    return asLongLow(traceId);
   }
 
-  /** Convert the "high bytes" of the given hex traceId into a long representation. */
-  public static long traceIdHighBytesAsLong(CharSequence traceId) {
+  /**
+   * Returns the "high part" of the {@code long} values representation of the {@code TraceId}
+   * converted from the given lowercase hex (base16) representation.
+   *
+   * @param traceId the lowercase hex (base16) representation of the {@code TraceId}.
+   * @return the {@code long} value representation of the {@code TraceId}.
+   * @throws NullPointerException if {@code traceId} is null.
+   */
+  public static long asLongHigh(CharSequence traceId) {
     return BigendianEncoding.longFromBase16String(traceId, 0);
   }
 
-  /** Convert the "low bytes" of the given hex traceId into a long representation. */
-  public static long traceIdLowBytesAsLong(CharSequence traceId) {
+  /**
+   * Returns the "low part" of the {@code long} values representation of the {@code TraceId}
+   * converted from the given lowercase hex (base16) representation.
+   *
+   * @param traceId the lowercase hex (base16) representation of the {@code TraceId}.
+   * @return the {@code long} value representation of the {@code TraceId}.
+   * @throws NullPointerException if {@code traceId} is null.
+   */
+  public static long asLongLow(CharSequence traceId) {
```

Comment:
Using `as` for longLow/High is misleading IMHO. This is not the trace ID "as" a longLow. It is only a part of the trace ID. Maybe `getLongLow`  would be better.

Reply:
`getLowAsLong()` is accurate, but might be a bit wordy?

Reply:
`lowAsLong()` ?

Reply:
I would try to make an exercise and write them in code:

```java
TraceId.asLongLow("12345678123456781234567812345678")
TraceId.getLongLow("12345678123456781234567812345678")
TraceId.getLowAsLong("12345678123456781234567812345678")
TraceId.lowAsLong("12345678123456781234567812345678")
TraceId.asLongLowPart("12345678123456781234567812345678")
TraceId.lowPartAsLong("12345678123456781234567812345678")
```

```java
SpanId.asLong("1234567812345678")
```

So far these were the suggestions, any recommendation based on this?

Reply:
For consistency reason I kind of like `asLongLowPart` or `asLongLow`, but I also see a bit the fact that it is not converting the entire id, only extracts one of the parts. But on the other hand the returned value represents the LowPart of the id

Reply:
I like `lowPartAsLong()` the best of those options.

Reply:
Now that I read it I feel the same. Sounds more like English and also describes what it happens

Reply:
Changed 👍